icu4j/main/core/src/main/java/com/ibm/icu/message2/InputSource.java

```diff
@@ -62,10 +62,22 @@ int readCodePoint() {
         return c;
     }
 
-    // Backup a number of characters.
+    // Backup a number of code points.
     void backup(int amount) {
+        int amountBackedUp = 0;
         // TODO: validate
-        cursor -= amount;
+        while (amountBackedUp < amount) {
+            backupOneCodePoint();
+            amountBackedUp++;
+        }
+    }
+
+    void backupOneCodePoint() {
```

Comment:
What happens if it is an unpaired low surrogate?
There is nothing to prevent such a thing from showing up in a message.

True, it is incorrect.
But we are not trying to reject that, we accept it.
And this code would misbehave.

Reply:
Sounds like we need a test for this! (I'll add one.)